### PR TITLE
Add functionality for activating/deactivating users in the users list

### LINKED ISSUE #25: Allow users to be toggled active and inactive from the /users index route
It would be great to be able to toggle users active and inactive directly from the /users table, so that we don't have to manually update via the users specific show route.

### PR DESCRIPTION
**Pull request by Serneum**
Closes #25

### CODE REVIEW

File: test/controllers/user_controller_test.exs [hunk 1]
```diff
@@ -63,4 +63,28 @@ defmodule Pairmotron.UserControllerTest do
...
+    user = Repo.insert! %User{name: "foo", email: "bar", active: true}
+    conn = get conn, user_path(conn, :index)
+    assert html_response(conn, 200) =~ "Active"
+
+    conn = put conn, user_deactivate_path(conn, :deactivate, user)
+    assert redirected_to(conn) == user_path(conn, :index)
+
+    conn = get conn, user_path(conn, :index)
+    assert html_response(conn, 200) =~ "Inactive"
+  end
```
Comment (mbramson) [inline test/controllers/user_controller_test.exs:89]: I think that these tests are testing too many things at once. You've even left whitespace where these tests should probably be broken out.

File: web/controllers/user_controller.ex [hunk 2]
```diff
@@ -38,28 +38,49 @@ defmodule Pairmotron.UserController do
...
+    |> redirect(to: user_path(conn, :index))
+  end
+
+  def activate(conn, %{"user_id" => id}) do
+    update_user(conn, id, %{active: true}, :index)
+  end
+
+  def deactivate(conn, %{"user_id" => id}) do
+    update_user(conn, id, %{active: false}, :index)
+  end
```
Comment (mbramson) [inline web/controllers/user_controller.ex]: We should definitely explore ways to accomplish this without adding new non-RESTful controller methods, as discussed.

**Review by mbramson — CHANGES_REQUESTED** [review]

**Comment by Serneum** [comment]
@mbramson Commit one isn't RESTful, but cleaner in my opinion. Commit two is the more RESTful but I hate it. Tell me which you'd prefer.

File: web/router.ex [hunk 3]
```diff
@@ -17,7 +17,10 @@ defmodule Pairmotron.Router do
     pipe_through :browser # Use the default browser stack
 
     get "/", PageController, :index
-    resources "/users", UserController
+    resources "/users", UserController do
+      put "/activate", UserController, :activate, as: :activate
+      put "/deactivate", UserController, :deactivate, as: :deactivate
```
Comment (daveshah) [inline web/router.ex:22]: This doesn't seem to follow RESTful type behavior. Why are 'activate' and 'deactivate' their own routes here?

**Comment by mbramson** [comment]
Closing in favor of https://github.com/ElixirCLE/pairmotron/issues/85, since the /users route no longer exists. That issue is a great opportunity to introduce react into this project.

**Pull request closed without merging**